### ISSUE TITLE
Expandable control in toolbar: usability issue

### ISSUE THREAD

**Issue by terezanovotna** [patternfly/patternfly-design#1071]
I came across this pattern in Insights, and I would love the team to reconsider it as it seems to me we are introducing an anti-pattern, which concerns me. 

PF Component:
https://www.patternfly.org/v4/components/data-list/react-demos/#expandable-control-in-toolbar

**Issues:**
- considering the toolbar is the primary action in our UI, having it as a primary action is unhappy location
- from researching other toolbars, the expansion action as any other action is usually towards the end of the toolbar
- I see this as antipattern

**Current behavior:**
Collapsed
<img width="1778" alt="Screen Shot 2021-07-21 at 10 16 32" src="https://user-images.githubusercontent.com/36040135/126455587-07fdc01f-d8c6-4163-a0b8-8fddf6ea2a3c.png">

Expanded
<img width="1785" alt="Screen Shot 2021-07-21 at 10 16 39" src="https://user-images.githubusercontent.com/36040135/126455612-3ae1f39f-a5ee-4fd5-baf9-5910b4c6fec2.png">

**Design proposal:**
change the location of the expandable control to less important
![expandable column](https://user-images.githubusercontent.com/36040135/126458626-c024abee-1e68-4b2b-941d-c48c71192e82.png)

Proposal 1
include it into kebab action behind export icon

Proposal 2
move this action to table header

**Comment by katierik** [comment]
I'm a big fan of Proposal 2 - Moving it to the table header. I think this would also work fine when the bulk select is in the table header as the bulk select can be aligned with the checkboxes and the expand all can be aligned with the expansion arrows.

**Comment by doruskova** [comment]
Agree! And it takes a lot of space! I'm a huge fan of version 2

**Comment by mmenestr** [comment]
@mcarrano - wanted to point your attention to this issue! To refresh your memory, my original design had the expand all icon in the header (just like they are asking for here), but we had decided to have the "expand all" button placed in the toolbar because you had expressed wanting the data list and table to have the same design solution for expand all. 
But I agree with them that it looks weird in this case, especially without the bulk select, and that it might be confused as an "expand toolbar".

Would you be willing to allow for both options for tables (ie: either put the expansion toggle in the header OR put it in the toolbar)? Interested to hear your thoughts!

**Comment by mcarrano** [comment]
@mmenestr I know that we had explored some of these different options before arriving at the current recommendation. Didn't we do usability testing on this or am I mistaken? I think there were problems with Proposal 2, but I can't remember what they were. I would need to find the history on this. Proposal 1 would certainly work, but there might be concerns about visibility.

I'm not opposed to changing this, but would only want to do so after considering all the pros and cons since I know this was something arrived at after considering the various options. As for consistency with Data List, I still think they should be consistent so whatever we decide must work for both IMO.

**Comment by mmenestr** [comment]
@mcarrano you're right that we did conduct a test, but the test was testing using an icon in the header of a table, vs. using an action button with the words "expand all" at the toolbar level. The test did not include a version where the expansion icon was in the toolbar. Here are [the test results](https://docs.google.com/presentation/d/13IVeSebbUziANv54Xyxv-2q8Om2OOWd1mxWNMG-2j3E/edit#slide=id.p) for reference.

I just dug up [the original issue](https://github.com/patternfly/patternfly-design/issues/905) where we discussed the ultimate design, and I don't think there were problems with proposal 2 other than it not being possible for data lists since data lists don't have headers, which lead to the toolbar design.

The big pro of keeping it in the toolbar is that it would go hand in hand with the bulk select, the negative is that it could be confused for an expansion of the toolbar (something you had also talked about in our original discussion), and that it looks kind of out of place when there is no bulk select, and there are chips present.

**Comment by terezanovotna** [comment]
thanks for finding all the links, I see there was a lot of work done. From what I am reading, I think Proposal 2 could work as an alternative to the current design. Happy to chip in my thoughts if you are going to discuss it on the PF call @mcarrano

**Comment by katierik** [comment]
Proposal 2 is also the most similar to the version which was tested & the proposal that came out of the test (Slide 19).

**Comment by mcarrano** [comment]
What if we just replaced the icon for expand all? Yes, we could move the tool to the table header, but I think that the fundamental problem is that the icon so firmly implies that the thing to the right is expandable. I looked at some other options. Existing in PF today we have 

`fa-expand` ![Screen Shot 2021-08-19 at 4.55.48 PM.png](https://images.zenhubusercontent.com/5ce816524e6c594dc1b0df4d/6979b9c7-d6bd-4b52-b3e0-9c97d7291e9b)

`fa-expand-arrows-alt` ![Screen Shot 2021-08-19 at 4.56.39 PM.png](https://images.zenhubusercontent.com/5ce816524e6c594dc1b0df4d/b8297def-78de-41a7-9874-71b3caa74a40)

I also think that `fa-angle-double-down` would work well ![Screen Shot 2021-08-19 at 4.57.35 PM.png](https://images.zenhubusercontent.com/5ce816524e6c594dc1b0df4d/b283886d-cfb7-4e53-b7f2-1d77ffd027b1)

@mceledonia would also love to get your thoughts.

**Comment by mmenestr** [comment]
I had already looked at using that first icon, but we concluded it would be confusing - to me it implies more of a "expand screen" than expand role, and same with the second icon.

I think the double arrow would work great though if we do change directions with the icon. I think it would even help clarify it as an expand all icon if we keep it in the toolbar + the study we conducted was with a double arrow icon.

**Comment by mcarrano** [comment]
After giving this some thought, I agree that we should proceed with implementing Proposal 2 as shown above. I think the better usability this affords in worth the inconsistency it creates between data lists and tables. I've opened https://github.com/patternfly/patternfly/issues/4366 and will get this on our roadmap. Thanks for raising this issue @terezanovotna .
